### ISSUE TITLE
Training on single speaker (male) hindi dataset - unable to attend (flow=1)

### ISSUE THREAD

**Issue by astricks** [NVIDIA/flowtron#53]
This is somewhat related to https://github.com/NVIDIA/flowtron/issues/39 but opening a new issue since I my model is unable to attend even with 1 flow.

My issue some also somewhat similar to https://github.com/NVIDIA/flowtron/issues/41 - I have now trained my model for 790,000 steps. Validation loss seems to have hit a minima at around 360k steps at which point attention was biased, and further training made attention vanish and validation loss slowly increase.

Attached below are attention plots for steps=215k, 360k, 790k; and the validation loss.
<img width="732" alt="attn215k" src="https://user-images.githubusercontent.com/1024213/89203172-1ef41980-d582-11ea-952c-57efe6b616fa.png">
<img width="692" alt="Attn360k" src="https://user-images.githubusercontent.com/1024213/89203137-0c79e000-d582-11ea-90a7-4ff8a26535d5.png">
<img width="741" alt="attn790k" src="https://user-images.githubusercontent.com/1024213/89203190-25829100-d582-11ea-8638-d1629aab61f3.png">
<img width="392" alt="valLoss" src="https://user-images.githubusercontent.com/1024213/89203208-2ddacc00-d582-11ea-9807-ca44460e6d8f.png">

I am wondering how to proceed. The options I'm considering are:
1. increase flow=2 and warmstart with checkpoint 790k.
2. increase flow=2 and warmstart with checkpoint 360k. I deleted that checkpoint to save some diskspace and am now regretting it - I'll have to restart training and get to 360k steps again.
3. Train a new tacotron2 model using my hindi dataset and warmstart flowtron using that.

@rafaelvalle Would love some advice at this point.

Also attaching some inference files below. The speech is senseless though.
![sid0_sigma0 5_attnlayer0](https://user-images.githubusercontent.com/1024213/89203567-c5d8b580-d582-11ea-8622-2b61c1526b4e.png)
[sid0_sigma0.5.wav.zip](https://github.com/NVIDIA/flowtron/files/5017062/sid0_sigma0.5.wav.zip)

**Comment by astricks** [comment]
Update: I trained tacotron2 with the same data - got good alignment and speech generation. Used that model to warmstart flowtron, but no got very little attention even at 500k steps. Decided to warmstart on LJS, hoping I get better results with that.

**Comment by astricks** [comment]
Warmstarting on LJS didn't work. I cleaned my dataset again, removing silences from start and end. Trying warmstarting off Tacotron model again.

**Comment by astricks** [comment]
An update, since this issue has been open for a long time. My model learned to attend, kinda. It still has issues during inference and I've been playing around with the inference config, but at least I'm seeing some attention.

Solution:
1. I created a new 3-speaker dataset
2. I cleaned it diligently, removing all silence from the beginning and end.
3. I also removed long silences from the middle of the audio clips.
4. CAVEAT: The transcriptions of the audio are not 100% accurate. There are a few wrong but similar-sounding transcriptions.

All three speakers have around 14 hours of data, yet some speakers attend better than others.

Here are the graphs, and some [samples](https://drive.google.com/drive/folders/1jYR-WJ-WRt6lE0HwWjxZcFznV2szkZXY?usp=sharing).

<img width="667" alt="Screen Shot 2020-10-05 at 10 07 47 AM" src="https://user-images.githubusercontent.com/1024213/95090009-cb718980-06f2-11eb-8472-41245b44c78d.png">
<img width="705" alt="Screen Shot 2020-10-05 at 10 08 15 AM" src="https://user-images.githubusercontent.com/1024213/95090041-d3c9c480-06f2-11eb-8290-2fc2e5d9c4e8.png">
<img width="656" alt="Screen Shot 2020-10-05 at 10 07 24 AM" src="https://user-images.githubusercontent.com/1024213/95090063-d88e7880-06f2-11eb-8fa6-68695c4e93b6.png">
<img width="347" alt="Screen Shot 2020-10-05 at 10 08 35 AM" src="https://user-images.githubusercontent.com/1024213/95090077-de845980-06f2-11eb-841d-3100ab70100d.png">


I'm now using my model from 920k iterations to warmstart a 2flow model. Hoping it improves attention and the quality of inference.

**Comment by rafaelvalle** [comment]
@astricks thank you for letting us know cleaning the data has helped the model learn attention. 
Can you please share how the attention looks at 200k iterations?

**Comment by astricks** [comment]
@rafaelvalle attaching attention plot for earlier iterations

<img width="546" alt="Screen Shot 2020-10-05 at 5 46 37 PM" src="https://user-images.githubusercontent.com/1024213/95135681-cb917980-0732-11eb-9baa-e7e7573b0da9.png">
<img width="555" alt="Screen Shot 2020-10-05 at 5 46 24 PM" src="https://user-images.githubusercontent.com/1024213/95135693-d0eec400-0732-11eb-930e-6c068bbba9a8.png">
<img width="529" alt="Screen Shot 2020-10-05 at 5 48 53 PM" src="https://user-images.githubusercontent.com/1024213/95135848-114e4200-0733-11eb-8b85-824c5d116a71.png">

**Comment by rafaelvalle** [comment]
Great. I suggest resuming from the model with 200k iters given that it has better generalization loss and less bias in the attention map.

**Comment by astricks** [comment]
Gotcha, thanks! Just warmstarted with 200k iteration checkpoint and n_flows=2.

**Comment by tricky61** [comment]
> An update, since this issue has been open for a long time. My model learned to attend, kinda. It still has issues during inference and I've been playing around with the inference config, but at least I'm seeing some attention.
> 
> Solution:
> 
> 1. I created a new 3-speaker dataset
> 2. I cleaned it diligently, removing all silence from the beginning and end.
> 3. I also removed long silences from the middle of the audio clips.
> 4. CAVEAT: The transcriptions of the audio are not 100% accurate. There are a few wrong but similar-sounding transcriptions.
> 
> All three speakers have around 14 hours of data, yet some speakers attend better than others.
> 
> Here are the graphs, and some [samples](https://drive.google.com/drive/folders/1jYR-WJ-WRt6lE0HwWjxZcFznV2szkZXY?usp=sharing).
> 
> <img alt="Screen Shot 2020-10-05 at 10 07 47 AM" width="667" src="https://user-images.githubusercontent.com/1024213/95090009-cb718980-06f2-11eb-8472-41245b44c78d.png">
> 
> <img alt="Screen Shot 2020-10-05 at 10 08 15 AM" width="705" src="https://user-images.githubusercontent.com/1024213/95090041-d3c9c480-06f2-11eb-8290-2fc2e5d9c4e8.png">
> 
> <img alt="Screen Shot 2020-10-05 at 10 07 24 AM" width="656" src="https://user-images.githubusercontent.com/1024213/95090063-d88e7880-06f2-11eb-8fa6-68695c4e93b6.png">
> 
> <img alt="Screen Shot 2020-10-05 at 10 08 35 AM" width="347" src="https://user-images.githubusercontent.com/1024213/95090077-de845980-06f2-11eb-841d-3100ab70100d.png">
> 
> I'm now using my model from 920k iterations to warmstart a 2flow model. Hoping it improves attention and the quality of inference.

how did you trim silence of the start and end? liborosa.trim or other method?
have you got attention with only ljs corpus? In the paper, 3 dataset also used to attention with n_flow=1, then n_flow=2. However when I only ues the ljs corpus, I cann't get attention when n_flow=1, even warm start from tacotron2 model which has good attention, and I use librosa to trim silence with top_db=30
